### PR TITLE
Support Demo Mode in Workflows

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

This enables Workflows to be run in demo mode without blowing up; it requires some additional env config also; the new keys are reflected in the `.env-cmd` example file and have been added to 1Password; they are identical to the regular development or staging settings though since the demo backend runs in the same GCP project.

You can't have multiple Firestore instances or databases in a single project, unfortunately, so we can't isolate the demo data as much as we might like to without, e.g., reading some extra metadata out of Auth0 or something. Didn't seem necessary at this point but if that is a big security concern I can look into it!

The demo collections do not as yet have any data in them, so you cannot actually use Workflows for anything useful, but I validated that it will work by loading some real data into them (and then deleting it!). We ought to be able to populate them as needed to support demo use cases. 

## Type of change

- [x] New feature (non-breaking change that adds functionality)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> #1635 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- ~E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))~

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/firestore/firestore.ts [hunk 1]
```diff
@@ -76,35 +91,56 @@ export const authenticate = async (
...
 const collections = {
-  staff: collection(db, "staff") as CollectionReference<StaffRecord>,
+  staff: collection(
+    db,
+    collectionNames.staff
+  ) as CollectionReference<StaffRecord>,
   userUpdates: collection(
     db,
-    "userUpdates"
+    collectionNames.userUpdates
```
Comment (Catacola) [inline src/firestore/firestore.ts:127]: Even without needing to support demo mode, this is much cleaner.

**Review by Catacola — APPROVED** [review]
Thanks for the rapid turn-around!

**Review by jovergaag — COMMENTED** [review]
I'm not seeing any officers or clients in demo mode. I thought there were a handful of clients in the fixture?

**Review by jovergaag — APPROVED** [review]
Great! I ran local demo mode and dev mode and everything looks good!

There are clearly some empty Pathways charts in demo mode, but that is (I think) because of the old fixtures in the demo bucket. Once you merge this, I will focus on getting Pathways all cleaned up and deploy the demo app

**Pull request merged**